File: chunlin/121.java
class Solution {
    public int maxProfit(int[] prices) {
        if (prices == null || prices.length < 2) {
            return 0;
        }
        int maxPro = 0;
        int min = prices[0];
        for (int i = 1; i < prices.length; i++) {
            maxPro = Math.max(maxPro, prices[i] - min);
            min = Math.min(min, prices[i]);
        }
        return maxPro;
    }
}

File: zemiao/121.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        if len(prices) < 2:
            return 0
        
        lowest, maxGain = prices[0], 0
        for price in prices:
            lowest = min(lowest, price)
            maxGain = max(maxGain, price-lowest)
        
        return maxGain